### PR TITLE
YAML truncates trailing floating point zeroes

### PR DESCRIPTION
**Pull request by Saransh-cpp**
<details>
<summary><strong>Instructions</strong></summary>

Thanks for contributing! :heart:

Keep in mind that **lesson maintainers are volunteers** and it may take them some time to respond to your contribution.
To ensure that the right people get notified, you can tag some of the last contributors with `@githubname`.

Although not all contributions can be incorporated into the lesson materials, we appreciate your time and effort to improve the curriculum. If you have any questions about the lesson maintenance process or would like to volunteer your time as a contribution reviewer, please contact the HSF training convenors ([contacts here](https://hepsoftwarefoundation.org/workinggroups/training.html)).

You may delete these instructions from your pull request.

\- HSF Training
</details>

Syncing with https://github.com/hsf-training/hsf-training-cicd-github/pull/58

cc: @klieret @wdconinc

### CODE REVIEW

File: _episodes/03-understanding-yaml.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ YAML (Yet Another Markup Language or sometimes popularly referred to as YAML Ain
 
 ~~~
 number-value: 42
-floating-point-value: 3.141592
```
Comment (klieret) [inline _episodes/03-understanding-yaml.md]: bit nitpicky, but this is what I would expect. For a floating point `1.10 == 1.1`. The issue with the python versions is that they are _not_ floating numbers yet are specified as such. So let's perhaps leave this line as it is and add another line "python-verion: 3.10" and then add the callout for this?

**Review by klieret — CHANGES_REQUESTED** [review]

File: _episodes/03-understanding-yaml.md (hunk 1 above)
Comment (wdconinc) [reply _episodes/03-understanding-yaml.md]: And even 1.1 is not actually 1.1 ;-) https://www.h-schmidt.net/FloatConverter/IEEE754.html.

I would be careful with putting line in the example listing which have incorrect behavior, even if we call attention to it later. If we put in a python-version line, we should do it was a string `'3.10'`, not as a float.

Comment (Saransh-cpp) [reply _episodes/03-understanding-yaml.md]: Thanks for the review! I've pushed another commit with the required modifications.

**Review by klieret — APPROVED** [review]
Perfect :)

**Comment by klieret** [comment]
@all-contributors please add @Saransh-cpp for content

**Comment by Saransh-cpp** [comment]
A gentle bump on this :)

I think this should be ready for merging.

**Pull request merged**